src/GPU3D.h

```diff
@@ -137,14 +137,12 @@ void Write32(u32 addr, u32 val);
 class Renderer3D
 {
 public:
-    Renderer3D(bool Accelerated);
     virtual ~Renderer3D() {};
 
     Renderer3D(const Renderer3D&) = delete;
     Renderer3D& operator=(const Renderer3D&) = delete;
 
-    virtual bool Init() = 0;
-    virtual void DeInit() = 0;
+    bool IsValid() const noexcept { return Valid; }
```

Comment:
I don't like that it's possible to a such an invalid Renderer object lying around.

I'd prefer it if the InitRenderer functions would just be skipped from the frontend side in case Init fails.

Reply:
This is generally why constructors throw exceptions, but since I don't see any exceptions in the codebase I assumed that melonDS doesn't use them.

Alternatively, I could hide the constructor and change the class to be created with a function that returns a `unique_ptr`.

Are either of these solutions okay?

Reply:
I have addressed this in my latest series of commits.